utils/matrix/setup_matrix.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import argparse
+import os
+import shutil
+
+try:
+    from configparser import ConfigParser
+except ImportError:
+    from six.moves import configparser
+    import six
+
+    if six.PY2:
+        ConfigParser = configparser.SafeConfigParser
+    else:
+        ConfigParser = configparser.ConfigParser
+
+from shared import get_logger, check_call
+
+
+# build targets from
+# https://mpas-dev.github.io/compass/latest/developers_guide/machines/index.html#supported-machines
+all_build_targets = {
+    'anvil': {
+        ('intel', 'impi'): 'intel-mpi',
+        ('intel', 'openmpi'): 'ifort',
+        ('intel', 'mvapich'): 'ifort',
+        ('gnu', 'openmpi'): 'gfortran',
+        ('gnu', 'mvapich'): 'gfortran'},
+    'badger': {
+        ('intel', 'impi'): 'intel-mpi',
+        ('gnu', 'mvapich'): 'gfortran'},
+    'chrysalis': {
+        ('intel', 'impi'): 'intel-mpi',
+        ('intel', 'openmpi'): 'ifort',
+        ('gnu', 'openmpi'): 'gfortran'},
+    'compy': {
+        ('intel', 'impi'): 'intel-mpi'},
+    'cori-haswell': {
+        ('intel', 'mpt'): 'intel-nersc',
+        ('gnu', 'mpt'): 'gnu-nersc'},
+    'conda-linux': {
+        ('gfortran', 'mpich'): 'gfortran',
+        ('gfortran', 'openmpi'): 'gfortran'},
+    'conda-osx': {
+        ('gfortran-clang', 'mpich'): 'gfortran-clang',
+        ('gfortran-clang', 'openmpi'): 'gfortran-clang'}
+}
```

Comment:
No, this information is very specific to MPAS so doesn't belong in `mache`.  It could live in `compass` but then it wouldn't be available to this utility, which lives firmly outside of the compass package.  For now, this info only lives in the documentation besides this dictionary.